docs/book/src/jobs.md

```diff
@@ -0,0 +1,42 @@
+# Jobs
+
+This document intents to provide an overview over our jobs running via Prow and Github actions.
```

Comment:
let's also add a reference to this document from https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/CONTRIBUTING.md#contributing-a-patch